spec/lib/site_setting_extension_spec.rb

```diff
@@ -1048,6 +1077,43 @@ def self.translate_names?
       expect(SiteSetting.search_experience(theme_id: theme_2.id)).to eq("search_field")
     end
 
+    it "publishes changes across processes" do
+      MessageBus.on
+
+      settings_tss_instance_1 = new_settings(provider_local)
+      settings_tss_instance_2 = new_settings(provider_local)
+
+      settings_tss_instance_1.listen_for_changes = true
+      settings_tss_instance_2.listen_for_changes = true
+
+      settings_tss_instance_1.load_settings(File.join(Rails.root, "config", "site_settings.yml"))
+      settings_tss_instance_2.load_settings(File.join(Rails.root, "config", "site_settings.yml"))
+
+      settings_tss_instance_1.refresh!
+      settings_tss_instance_2.refresh!
+
+      expect(settings_tss_instance_1.enable_welcome_banner(theme_id: theme_1.id)).to eq(false)
+      expect(settings_tss_instance_2.enable_welcome_banner(theme_id: theme_1.id)).to eq(false)
+
+      result =
+        Themes::ThemeSiteSettingManager.call(
+          params: {
+            theme_id: theme_1.id,
+            name: :enable_welcome_banner,
+            value: true,
+          },
+          guardian: Discourse.system_user.guardian,
+        )
+
+      expect(result.success?).to eq(true)
+
+      # Get through the MessageBus queue
+      try_until_success(frequency: 0.5) do
```

Comment:
Not sure of a better way to handle this, there isn't a way to make MessageBus process all messages. Without this delay, the spec fails